### ISSUE TITLE
Can't see within .var folder

### ISSUE THREAD

**Issue by 35609902357** [flathub/org.freefilesync.FreeFileSync#47]
The application can only see it's own subfolder, all the others are treated as non-existent

**Comment by kparal** [comment]
`$HOME/.var/app` gets mounted as tmpfs and `$HOME/.var/app/org.freefilesync.FreeFileSync` gets bind-mounted from the original location by flatpak. I'm not sure if anything can be done to avoid that. Is that a problem?

**Comment by 35609902357** [comment]
It is a big problem. I accidentally found out all my Flatpak applications were not backed up. Now I have to use another backup program at least for the .var folder.

**Comment by kparal** [comment]
You can use the upstream FFS installer, and it will not have this limitation. FFS existence on Flathub is unsure anyway, see #48. If FFS stays on Flathub, I'm happy to accept patches to work around this, but I'm not even sure if it is possible.

**Comment by MikeNavy** [comment]
Hi, I use Thunderbird flatpak. All my mails are in my thunderbird profile, itself in "/home/.../.var/app/org.mozilla.Thunderbird/.thunderbird". When I read this issue, I realized my mails were not backed up. I uninstalled FFS flatpack and installed FFS from its website. Now my mails (3.2 GB!) are backed-up. I have also 13 other flatpak apps that were not backed up (and are now). You should understand that a backup software that doesn't do its job is useless; moreover, it becomes dangerous when you believe it does the job while it doesn't. I don't know why is the bug and how to correct it, but **without correction there is no future for FFS flatpak**.

**Comment by kparal** [comment]
@MikeNavy No need to get dramatic. Yes, if you want to back up your whole home, it is a serious disadvantage. But that's just one of the use cases of FFS (for example I don't use it that way). I'll add this information to the package description, so that people (at least those who read) can be informed.

We're still waiting for a volunteer to investigate this problem and figure out whether the access to those files can be enabled somehow.

**Comment by MikeNavy** [comment]
Hi, from flatpak reference documentation it seems that "/var" in the host filesystem is blacklisted:
"As mentioned above the host option does not actually provide complete access to the host filesystem. The main rules are:
These directories are blacklisted: /lib, /lib32, /lib64, /bin, /sbin, /usr, /boot, /root, /tmp, /etc, /app, /run, /proc, /sys, /dev, /var".

However, this does not say that "/.var" directory is also blacklisted in "/home".
- It might be a bug, to file in flatpak issues.
- Maybe, adding the permission to access "/home/**username**/.var" could solve the problem (permission can easily be tested using flatseal, and if it works, added in the manifest).

Regards,
MN

**Comment by kparal** [comment]
Yes, anyone who has time to experiment with this, please test using Flatseal to grant access to `~/.var` or `~/.var/app`. And if it doesn't work, please ask more knowledgeable people using one of the [communication channels](https://flatpak.org/about/). Thanks.

**Comment by mrzeld** [comment]
Recently got a steam deck. Installed EmuDeck on it and wanted to be able to backup my save files from the various emulators. Many of them are saved in the "/home/deck/.var/app/*" folder. FFS was unable to see these folders to backup the needed files. 

So based on this thread, I opened FlatSeal and modified FFS settings. In the "Other Files" section under Filesystem section (see image below), I individually added all the complete paths to the save folders within various apps located in the /.var/app folder. Now FFS can see these folders and back them up. 

Glad I found this thread, but would be nice if FFS could access these folders off the bat. For reference, I first downloaded an app called syncBackup, and it could access the files in the /.var/app folder. But FFS is MUCH NICER.  

Thanks again!

![image](https://user-images.githubusercontent.com/17414737/226075899-b667e5c8-1e4a-4f80-acb6-e19954cca82c.png)

**Comment by kparal** [comment]
> In the "Other Files" section under Filesystem section (see image below), I individually added all the complete paths to the save folders within various apps located in the /.var/app folder. Now FFS can see these folders and back them up.

Thanks, that's useful. I just tested that and it should be enough to just add a single path `/home/username/.var/app`, so e.g. `/home/deck/.var/app` in your case.

> I first downloaded an app called syncBackup, and it could access the files in the /.var/app folder.

Huh, this is mighty interesting. I can confirm it can see there, but only using the file open dialog. If I run `flatpak run --command=bash com.darhon.syncbackup`, I can't see there. So I assume this is some file-portal in action. FFS still uses old gtk2, but there was a recent commit (then reverted, #103) using gtk3, and when I tried it, it still didn't see into `~/.var/app`. Either gtk4 is needed, or the FFS author bundles some old gtk3 version, or he makes some incompatible modifications.
